pkg/monitor/cluster/hiveregistrationstatus_test.go

```diff
@@ -0,0 +1,270 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	hiveclient "github.com/openshift/hive/pkg/client/clientset/versioned"
+	hivefake "github.com/openshift/hive/pkg/client/clientset/versioned/fake"
+	"github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/hive"
+	mock_metrics "github.com/Azure/ARO-RP/pkg/util/mocks/metrics"
+)
+
+func TestEmitHiveRegistrationStatus(t *testing.T) {
+	fakeNamespace := "fake-namespace"
+
+	for _, tt := range []struct {
+		name       string
+		oc         *api.OpenShiftCluster
+		cd         kruntime.Object
+		withClient bool
+		wantErr    string
+		wantLog    string
+	}{
+		{
+			name:       "no hiveclient",
+			withClient: false,
+			wantLog:    "skipping: no hive cluster manager",
+		},
+		{
+			name:       "No Namespace",
+			withClient: true,
+			oc: &api.OpenShiftCluster{
+				Name: "testcluster",
+				Properties: api.OpenShiftClusterProperties{
+					HiveProfile: api.HiveProfile{
+						Namespace: "",
+					},
+				},
+			},
+			wantErr: "cluster testcluster not adopted. No namespace in the clusterdocument",
+		},
+		{
+			name:       "Happy Path",
+			withClient: true,
+			oc: &api.OpenShiftCluster{
+				Name: "testcluster",
+				Properties: api.OpenShiftClusterProperties{
+					HiveProfile: api.HiveProfile{
+						Namespace: fakeNamespace,
+					},
+				},
+			},
+			cd: &hivev1.ClusterDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      hive.ClusterDeploymentName,
+					Namespace: fakeNamespace,
+				},
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			var hiveclient hiveclient.Interface
+			if tt.withClient {
+				fakeclient := hivefake.NewSimpleClientset()
+				if tt.cd != nil {
+					fakeclient.Tracker().Add(tt.cd)
+				}
+				hiveclient = fakeclient
+			}
+
+			logger, hook := test.NewNullLogger()
+			log := logrus.NewEntry(logger)
+
+			mon := &Monitor{
+				hiveclientset: hiveclient,
+				oc:            tt.oc,
+				log:           log,
+			}
+
+			err := mon.emitHiveRegistrationStatus(context.Background())
+			if err != nil && err.Error() != tt.wantErr ||
+				err == nil && tt.wantErr != "" {
+				t.Error(err)
+			}
+
+			if tt.wantLog != "" {
+				x := hook.LastEntry()
+				assert.Equal(t, tt.wantLog, x.Message)
+			}
```

Comment:
Nice one. In this specific case, I wouldn't bother testing logs since it is a temporary `if` & log line and this test case will go away.

But in general it is a nice approach to testing logs. I think I was using something more weird to do that in the past.

Edit: just to clarify: since we already have it implemented - do not remove it.